tiny_dnn/parameter.h

```diff
@@ -0,0 +1,125 @@
+/*
+    Copyright (c) 2013, Taiga Nomi and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+#include "tiny_dnn/core/framework/tensor.h"
+#include "tiny_dnn/util/util.h"
+
+namespace tiny_dnn {
+
+enum class parameter_type : int8_t { weight = 0x0001, bias = 0x0002 };
+
+class parameter {
```

Comment:
@beru agree with you, however not sure how much work implies this feature since xtensor is templated. But I like the idea of dynamic type. /cc @Randl